Math/NumberTheory/Primes/Factorisation/LinearAlgebra.hs

```diff
@@ -30,6 +31,8 @@ instance Semigroup SBVector where
 instance Monoid SBVector where
   mempty = SBVector mempty
 
+dot :: SBVector -> SBVector -> Bit
+dot (SBVector v1) (SBVector v2) = if even (S.size (v1 `S.intersection` v2)) then Bit True else Bit False
 -- Sparse Binary Matrix
 newtype SBMatrix = SBMatrix (I.IntMap SBVector) deriving (Show)
```

Comment:
Since you never insert or remove rows, I think it would work faster with `Data.Vector.Vector` instead of `I.IntMap`.